File: react/src/modules/post/interfaces/post.response.ts
import { BaseResponse } from '../../../@core/interfaces'
import { PostBaseModel, ProductDetailModel } from './post.model'

export interface PostResponse extends BaseResponse {
	offset: number
	limit: number
	total: number
	totalPage: number
	currentPage: number
	response: PostBaseModel[]
}

export interface PostIdResponse extends BaseResponse {
	response: ProductDetailModel
}

File: shop-admin-backend/src/main/java/com/demo/admin/modules/post/responses/PostResponse.java
package com.demo.admin.modules.post.responses;

import java.util.List;

import com.demo.admin.modules.post.models.PostModel;

import lombok.Builder;
import lombok.Data;

@Data
@Builder
public class PostResponse {
    int err;
    String msg;
    int total;
    int totalPage;
    int currentPage;
    int offset;
    int limit;
    List<PostModel> response;
}
